l3build-check.lua

```diff
@@ -992,32 +1007,36 @@ function save(names)
     return 1
   end
   for _,name in pairs(names) do
-    if testexists(name) then
+    local test_filename = testexists(name)
+    if test_filename then
       for _,engine in pairs(engines) do
         local testengine = ((engine == stdengine and "") or "." .. engine)
-        local function save_test(test_ext,gen_ext,out_ext,pdfmode)
-          local out_file = name .. testengine .. out_ext
-          local gen_file = name .. "." .. engine .. gen_ext
+        local function save_test(test_type)
+          local out_file = name .. testengine .. test_type.reference
+          local gen_file = name .. "." .. engine .. test_type.generated
           print("Creating and copying " .. out_file)
-          runtest(name,engine,false,test_ext,pdfmode)
-          ren(testdir,gen_file,out_file)
-          cp(out_file,testdir,testfiledir)
-          if fileexists(unpackdir .. "/" .. out_file) then
-            print("Saved " .. out_ext
+          runtest(name, engine, false, test_type.test, test_type)
+          ren(testdir, gen_file, out_file)
+          cp(out_file, testdir, testfiledir)
+          if fileexists(unpackdir .. "/" .. test_type.reference) then
+            print("Saved " .. test_type.reference
               .. " file overrides unpacked version of the same name")
             return 1
           end
           return 0
         end
         local errorlevel
-        if fileexists(testfiledir .. "/" .. name .. lvtext) then
-          errorlevel = save_test(lvtext,logext,tlgext)
-        else
-          errorlevel = save_test(pvtext,pdfext,tpfext,true)
+        for _, kind in ipairs(test_order) do
+          local test_type = test_types[kind]
+          if test_filename == testfiledir .. "/" .. name .. test_type.test then
+            errorlevel = save_test(test_type)
+            break
+          end
         end
-        if errorlevel ~=0 then return errorlevel end
+        if errorlevel ~= 0 then return errorlevel end
       end
     elseif locate({unpackdir, testfiledir}, {name .. lveext}) then
+      -- FIXME: This doesn't look like it does what it claims to do.
       print("Saved " .. tlgext .. " file overrides a "
         .. lveext .. " file of the same name")
       return 1
```

Comment:
I left this error as is for now because I didn't understand what it's supposed to do. It errors if no test file, but an expectation file exists. But why is it talking about overwriting? If I'm not missing something, the saved tlgfile can't overwrite anything because without the testfile we can't even generate a tlg file.

Reply:
This was written a while ago when things were rather less complicated: I may well have copied the text between two places where the outcomes where different! I agree it doesn't look like it doesn't seem to make sense: change the text to say that?

Reply:
I would just drop this branch. It only triggers when a `lve` file exists but no test and then `l3build save` is called. While the `lve` file is odd, the real issue we should be reporting is the missing test.

Reply:
@josephwright The new version now moves the error to actually warn if a lve is present while trying to save a test.